docs/spec/consensus/fork-accountability.md

```diff
@@ -0,0 +1,142 @@
+a block B. Note that in this case +1/3 of faulty validators does not need to commit an equivocation as they only vote once per round. Detecting faulty validators in case of such attack require fork accountability mechanism described in 
+this document https://docs.google.com/document/d/11ZhMsCj3y7zIZz4udO9l25xqb0kl7gmWqNpGVRzOeyY/edit?usp=sharing. 
+
+If a lite client is attacked using this attack with +1/3 of voting power, attacker can not arbitrarily change application state. In case there is an attack with +2/3 of voting power, an attacker can arbitrarily change application state. Note that in this case at least +1/3 of faulty validators locked value A in round r, so if they sign different value in follow up rounds this will be detectable by the the fork accountability mechanisms. The remaining 1/3 of faulty processes might not locked value A in round r so they can't be detected using this mechanism. Only in case they signed something which conflicts with the application this can be used against them. Otherwise they don't do anything incorrect. Note that this case is not covered by the report https://docs.google.com/document/d/11ZhMsCj3y7zIZz4udO9l25xqb0kl7gmWqNpGVRzOeyY/edit?usp=sharing as it only assumes at most 2/3 of faulty validators.
+
+Q: do we need to define a special kind of attack for the case where a validator sign arbitrarily state? It seems that
+detecting such attack requires different mechanism that would require as an evidence a sequence of blocks that lead to
+that state. This might be very tricky to implement. 
+
+### Back to the past
+
+In this kind of attacks faulty validators take advantage of the fact that he hasn't signed messages in some 
+of the past rounds. Due to asynchronous network in which Tendermint operate we cannot easily differentiate between
+such an attack and delayed message. This kind of attack can be used at both full nodes and lite clients.
+
+Example scenario: in a round r of height h we have 1/3 of correct validators precommitting a value A, 1/3 of correct
+validators precomit nil, 1/3 of faulty validators don't send any message and 1 faulty validator precommit nil.
+In some round r' > r, +1/3 of faulty validators together with 1/3 of correct that haven't locked any value commit some other value B. +1/3 of faulty validators now "go back to the past" and sign precomit message for value A in round r.
+Together with precomit messages of correct processes this for a commit for value A. Note that only a faulty validator
+that previously precomit nil did equivocation, while other 1/3 of faulty validators actually executed an
+attack that has exactly the same sequence of messages as part of amnesia attack. Detecting this kind of attack boil down to mechanisms for equivocation and amnesia. 
+
+Q: should we keep this as a separate kind of attacks? It seems that equivocation, amnesia and fantom validators 
```

Comment:
I think it would make sense to distinguish the case where 
(1) correct validators disagree on a block
(2) all correct validators agree, but some messages from the past are used to forge headers.

The back to the past, as I understand it, is of type (2). To me it seems less harmful, because it should be clear which is the correct block of a given height. If we have equivocation and two correct validators decide on different blocks, I think we need social consensus to decide which one was the right one (if any).